website/docs/contributing/ADRs/front-end/jsx-conditionals.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "ADR: Use of conditionals in JSX (deprecation of `<ConditionallyRender />`)"
+---
+
+## Background
+
+Using the `&&` operator in React can lead to unexpected rendering behavior when dealing with certain falsy values. In our codebase, the `<ConditionallyRender />` component has been used to render React elements based on a boolean condition. However, it has certain drawbacks, which is why we would like to replace it with the ternary operator.
+
+### Pitfalls of `&&` operator
+
+While most truthy and falsy values behave as expected with the `&&` operator, certain falsy values can produce unintended outcomes:
+
+```tsx
+{NaN && <p>❔</p>} // will render `NaN`
+{0 && <p>❔</p>} // will render `0`
+{arr?.length && <p>❔</p>} // can render `0`
+```
+
+These issues can cause bugs in components that conditionally render UI elements based on numeric values or other potentially falsy conditions. For this reason, we use a wrapper.
+
+```tsx
+<ConditionallyRender
+    condition={arr?.length}
+    show={<p>❔</p>}
+/>
+```
+
+### What's wrong with `<ConditionallyRender />`
+
+While this solves leaking render issues, it has some drawbacks.
+
+#### Poor TypeScript support
+```tsx
+import type { FC } from 'react';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+
+const SubComponent: FC<{ text: string }> = ({ text }) => <>{text}</>;
+
+export const Test: FC<{ maybeString?: string }> = ({ maybeString }) => (
+    <ConditionallyRender
+        condition={maybeString}
+        // ❌ TS Error: Type 'string | undefined' is not assignable to type 'boolean'
+        // You have to use `Boolean(maybeString)`
+
+        show={<SubComponent text={maybeString} />}
+        // ❌ TS Error: Type 'string | undefined' is not assignable to type 'string'
+        // you have to use `maybeString!` here
+    />
+);
+```
+
+#### Obfuscation of code smells and code cruft
+Nested ternaries are easier to spot than nested `<ConditionallyRender />` elements.
+
+```tsx
+<div>
+    <ConditionallyRender
+        condition={Boolean(a)} 
+        show={(
+            <ConditionallyRender
+                condition={Boolean(b)}
+                show={<p>This is bad</p>}
+            />
+        )}
+        elseShow={'Should be refactored'}
+    />
+</div>
+```
```

Comment:
Would have been nice to see the suggested approach instead, so we could compare:

```tsx
<div>
    {Boolean(a) ? (
        Boolean(b) && <p>This is bad</p>
    ) : 'Should be refactored'}
</div>
```

Reply:
https://github.com/Unleash/unleash/pull/8063